### DISCUSSION TITLE
How can I setup neotree to preview/open files like VSCode?

### DISCUSSION THREAD

**Discussion by julien-h2** [nvim-neo-tree/neo-tree.nvim#1557] [Q&A]
Hello, I need help to setup neotree to behave like the file explorer in VSCode. I will try to explain the best I can:

```
VSCode has a notion of preview-tab and normal-tab

1. If there is no preview-tab:
    click on a file in the tree ---> open new tab and mark as preview-tab then goto 2

2. If the current-tab is not a preview-tab:
    single click on a file in the tree ---> set preview-tab as current and preview file in it

2. If the current-tab is a preview-tab:
    single click on a file in the tree ---> preview file in preview-tab
    double click  on a file in the tree ---> open file in preview-tab and set the tab as normal-tab
    editing file in preview-tab ---> set the tab as normal-tab
```

**Comment by julien-h2** [comment]
So basically, I need toggle-preview to be always on and the behavior will be the same but it seems like neotree does not provide an option to do so?

**Comment by pynappo** [comment]
being tracked at #545, i'll see about implementing this over the weekend

**Reply by comod** [reply]
this would be exactly what i am looking for

**Comment by pynappo** [comment]
does this event handler do what you'd want?

https://github.com/nvim-neo-tree/neo-tree.nvim/issues/546#issuecomment-1259752317

Also, how often should neo-tree try to re-enable the preview? is it every time you move your cursor back into the neo-tree window, or everytime you open neo-tree back up?
